### PR TITLE
Remove join constraint check

### PR DESCRIPTION
**Pull request by miniway**

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/sql/planner/plan/JoinNode.java [hunk 1]
```diff
@@ -118,12 +118,6 @@ public JoinNode(
                     "%s join do not work with %s distribution type",
                     type,
                     distributionType.get());
-            // It does not make sense to PARTITION when there is nothing to partition on
-            checkArgument(
-                    !(distributionType.get() == PARTITIONED && criteria.isEmpty() && type != RIGHT && type != FULL),
-                    "Equi criteria are empty, so %s join should not have %s distribution type",
-                    type,
-                    distributionType.get());
         }
```
Comment (shimamoto) [inline presto-main/src/main/java/io/prestosql/sql/planner/plan/JoinNode.java:121]: According to https://github.com/trinodb/trino/pull/3245, maybe we should remove the following as well:
```java
import static io.prestosql.sql.planner.plan.JoinNode.DistributionType.PARTITIONED;
```

File: presto-parser/src/test/java/io/prestosql/sql/parser/TestSqlParserErrorHandling.java [hunk 2]
```diff
@@ -183,7 +183,7 @@ public void testPossibleExponentialBacktracking2()
                         "OR (f()\n" +
                         "OR (f()\n" +
                         "GROUP BY id",
-                "line 24:1: mismatched input 'GROUP'. Expecting: ')', ',', '.', 'FILTER', 'IGNORE', 'OVER', 'RESPECT', '['");
+                "line 24:1: mismatched input 'GROUP'. Expecting: ')', ',', '.', 'FILTER', 'OVER', '['");
```
Comment (shimamoto) [inline presto-parser/src/test/java/io/prestosql/sql/parser/TestSqlParserErrorHandling.java:186]: Is there any reason? It looks like there is no change in the original one.
https://github.com/trinodb/trino/blob/4bcd5b5c35cddd3c076a616282aaf57168236052/core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParserErrorHandling.java#L195

File: presto-main/src/test/java/io/prestosql/sql/query/TestExpressions.java [hunk 3]
```diff
@@ -34,21 +34,6 @@ public void teardown()
...
-        assertions.assertQuery("VALUES CASE 1 BETWEEN 0 AND 2 WHEN true THEN 10 ELSE 20 END", "VALUES 10");
-        assertions.assertQuery("VALUES CASE 1 NOT BETWEEN 0 AND 2 WHEN true THEN 10 ELSE 20 END", "VALUES 20");
-        assertions.assertQuery("VALUES CASE 1 IN (1, 2) WHEN true THEN 10 ELSE 20 END", "VALUES 10");
-        assertions.assertQuery("VALUES CASE 1 NOT IN (1, 2) WHEN true THEN 10 ELSE 20 END", "VALUES 20");
-        assertions.assertQuery("VALUES CASE 1 = 1 WHEN true THEN 10 ELSE 20 END", "VALUES 10");
-        assertions.assertQuery("VALUES CASE 1 = 2 WHEN true THEN 10 ELSE 20 END", "VALUES 20");
-        assertions.assertQuery("VALUES CASE 1 < 2 WHEN true THEN 10 ELSE 20 END", "VALUES 10");
-        assertions.assertQuery("VALUES CASE 1 > 2 WHEN true THEN 10 ELSE 20 END", "VALUES 20");
-    }
-
```
Comment (shimamoto) [inline presto-main/src/test/java/io/prestosql/sql/query/TestExpressions.java:51]: Is there any reason? It looks like there is no change in the original one.
https://github.com/trinodb/trino/blob/4bcd5b5c35cddd3c076a616282aaf57168236052/core/trino-main/src/test/java/io/trino/sql/query/TestExpressions.java#L40

**Review by shimamoto — COMMENTED** [review]
I added several comments. Could you take a look?

File: presto-parser/src/test/java/io/prestosql/sql/parser/TestSqlParserErrorHandling.java (hunk 2 above)
Comment (miniway) [reply presto-parser/src/test/java/io/prestosql/sql/parser/TestSqlParserErrorHandling.java:186]: This test fails as we haven't cherry-picked related changes

File: presto-main/src/test/java/io/prestosql/sql/query/TestExpressions.java (hunk 3 above)
Comment (miniway) [reply presto-main/src/test/java/io/prestosql/sql/query/TestExpressions.java:51]: This is the same, this test fails as we haven't cherry-picked related changes

File: presto-main/src/main/java/io/prestosql/sql/planner/plan/JoinNode.java (hunk 1 above)
Comment (miniway) [reply presto-main/src/main/java/io/prestosql/sql/planner/plan/JoinNode.java:121]: oh ok, I'm going to remove this

File: presto-parser/src/test/java/io/prestosql/sql/parser/TestSqlParserErrorHandling.java (hunk 2 above)
Comment (shimamoto) [reply presto-parser/src/test/java/io/prestosql/sql/parser/TestSqlParserErrorHandling.java:186]: Alright

File: presto-main/src/test/java/io/prestosql/sql/query/TestExpressions.java (hunk 3 above)
Comment (shimamoto) [reply presto-main/src/test/java/io/prestosql/sql/query/TestExpressions.java:51]: Alright

**Review by shimamoto — APPROVED** [review]
Looks good, thank you.

**Pull request merged**